src/util/fetch.ts

```diff
@@ -12,7 +13,16 @@ export const fetchAPIData = async (url: string, method: string, body:object , he
             ...headers,
         },
     })
-    const response = await data.json();
-    setCookie('shelf-share-session', JSON.stringify(body) , 1 )
-    return response; 
+    
+    if(response.status.toString().startsWith('4'||'5')){
```

Comment:
You can't put an || into a function like this. It always evaluate to '4' in this case since it is truthy. You would need to do response.status.toString().startsWith('4') || response.status.toString().startsWith('5').